doc/quickstart.rst

```diff
@@ -0,0 +1,196 @@
+       mainProxy = thisProxy;
+       arr[0].SayHi(0);
+     };
+
+     void done()
+     {
+       // Finish computation
+       CkPrintf("All done.\n");
+       CkExit();
+     };
+   };
+
+   /*array [1D]*/
+   class Hello : public CBase_Hello 
+   {
+   public:
+     Hello() {}
+
+     void SayHi()
+     {
+       // thisIndex stores the element’s array index 
+       CkPrintf("PE %d says: Hello world from element %d.\n", CkMyPe(), thisIndex);
+       if (thisIndex < nElem - 1) {
+         thisProxy[thisIndex + 1].SayHi(); // Pass the hello on
+       } else {
+         mainProxy.done(); // We've been around once -- we're done.
+       }
+     }
+   };
+
+   #include "hello.def.h" // created from hello.ci file above
+
+
+Compiling the Example
+'''''''''''''''''''''
+
+Charm++ has a compiler wrapper, ``charmc``, to compile Charm++ applications. Please see
+Section :numref:`sec:compile` for more information about ``charmc``.
+
+.. code-block:: console
+
+   $ charm/bin/charmc hello.ci # creates hello.def.h and hello.decl.h
+   $ charm/bin/charmc hello.cpp -o hello
+
+
+Running the Example
+'''''''''''''''''''
+
+Charm++ applications are started via ``charmrun``,
+which is automatically created by the ``charmc`` command above.
+Please see Section :numref:`sec:run` for more information about ``charmrun``.
+
+To run the application on two processors, use the following command:
+
+.. code-block:: console
+
+   $ ./charmrun +p2 ./hello
+   Charmrun> scalable start enabled.
+   Charmrun> started all node programs in 1.996 seconds.
+   Charm++> Running in non-SMP mode: 1 processes (PEs)
+   Converse/Charm++ Commit ID: v6.9.0-172-gd31997cce
+   Charm++> scheduler running in netpoll mode.
+   CharmLB> Load balancer assumes all CPUs are same.
+   Charm++> Running on 1 hosts (1 sockets x 4 cores x 2 PUs = 8-way SMP)
+   Charm++> cpu topology info is gathered in 0.000 seconds.
+   Running Hello on 2 processors with 8 elements.
+   PE 0 says: Hello world from element 0.
+   PE 0 says: Hello world from element 1.
+   PE 0 says: Hello world from element 2.
+   PE 0 says: Hello world from element 3.
+   PE 1 says: Hello world from element 4.
+   PE 1 says: Hello world from element 5.
+   PE 1 says: Hello world from element 6.
+   PE 1 says: Hello world from element 7.
+   All done
+   [Partition 0][Node 0] End of program
+
+
+Where to go From Here
+---------------------
+
+- The ``tests/charm++/simplearrayhello`` folder in the Charm++ distribution has a more comprehensive example, from
+  which the example in this file was derived.
+
+- The main Charm++ manual (https://charm.readthedocs.io/) contains more information about developing
+  and running Charm++ applications.
+
+- Charm++ has lots of other features, such as chare migration, load balancing,
+  and checkpoint/restart. The main manual has more information about them.
+
+- AMPI is an implementation of MPI on top of Charm++.
```

Comment:
Maybe add a link to the AMPI manual here? (https://charm.readthedocs.io/en/latest/ampi/manual.html)

Reply:
done